kerastuner/engine/tuner.py

```diff
@@ -89,26 +84,24 @@ def __init__(self,
                  logger=None,
                  tuner_id=None,
                  overwrite=False):
+
+        hypermodel = hm_module.KerasModelWrapper(
```

Comment:
Should we expose the class in the public API, and not wrap what was passed by the user if it's already a subclass of the Keras HyperModel? This would allow people to provide their own subclass when necessary.

Reply:
Done. Made it so that subclasses of `KerasHyperModel` are not wrapped. I think it's probably ok to leave this as `kt.engine.hypermodels.KerasHyperModel` rather than expose more publicly because the assumptions made by this class are very minimal and so subclassing will probably be niche

The only assumptions made are:

* A Keras model should be returned
* 5 tries at building the model should be allowed before crashing

Even the assumption that the Keras model be compiled is removed in this PR, since `fit` will already check this and users of custom training loops will not expect to need to compile